### PR TITLE
Add first terms to Glossary

### PR DESCRIPTION
**Pull request by tzhelyazkova**
Moving terms from the Google doc to the repo.

### CODE REVIEW

File: 12-Glossary.md [hunk 1]
```diff
@@ -1 +1,40 @@
...
+A textual description of the functionality provided by the system that captures actor-system interaction. It specifies how a user interacts with a system and how the system responds to the user actions.
+
+## Component
+
+A component is a software unit with a well-defined interface and explicitly specified dependencies. A software component can be as small as a block of reusable code, or it can be as big as an entire application depending on the level of granularity a system is being looked at.
+
+## MediaWiki Extension
+
+Extensions are software components that administrators can add (or remove) to their MediaWiki application in order to customize how it looks and works.
+Examples: WikibaseClient, UniversalLanguageSelector, OAuth
```
Comment (jakobw) [inline 12-Glossary.md:26]: What do you think about completely removing any text for terms and concepts that are well-defined elsewhere like MediaWiki concepts and Wikibase data model terminology, and only link to the respective site?

```suggestion
[https://www.mediawiki.org/wiki/Manual:Extensions](https://www.mediawiki.org/wiki/Manual:Extensions)
```

... and the same for sitelinks and interwiki links.

**Review by jakobw — COMMENTED** [review]
Nice, looks good!

I'm curious to hear others' thoughts on only linking to concepts defined elsewhere vs briefly describing them in our glossary too.

File: 12-Glossary.md (hunk 1 above)
Comment (jakobw) [reply 12-Glossary.md:26]: Ok, after a quick chat we agreed that brief descriptions are useful. That's fine with me, too!

File: 12-Glossary.md [hunk 2]
```diff
@@ -1 +1,40 @@
...
+
+## Use case
+
+A textual description of the functionality provided by the system that captures actor-system interaction. It specifies how a user interacts with a system and how the system responds to the user actions.
+
+## Component
+
+A component is a software unit with a well-defined interface and explicitly specified dependencies. A software component can be as small as a block of reusable code, or it can be as big as an entire application depending on the level of granularity a system is being looked at.
+
+## MediaWiki Extension
```
Comment (jakobw) [inline 12-Glossary.md:23]:
```suggestion
## [MediaWiki Extension](https://www.mediawiki.org/wiki/Manual:Extensions)
```

**Review by jakobw — APPROVED** [review]

**Review by Silvan-WMDE — APPROVED** [review]
Looks good so far

**Pull request merged**